Build broken with LuaJIT

Just to let you know, currently, both LuaJIT and luasocket define `luaL_setfuncs` which leads to link-time errors like 

```
/usr/local/lib/libluajit-5.1.a(lib_aux.o): In function `luaL_setfuncs':
/root/luajit-2.0/src/lib_aux.c:149: multiple definition of `luaL_setfuncs'
/usr/local/lib/lua/5.1/socket/libunix.a(compat.o):/root/luasocket/src/compat.c:7: first defined here
collect2: error: ld returned 1 exit status
```

Not sure if there is anything reasonable you can do about it though :(

Issue still applies. How to reproduce:

Build for Linux using the include path for your LuaJIT installation:
```
make MYCFLAGS="-I/home/siffiejoe/.self/programs/luajit-2.1/include/luajit-2.1" linux
```
Compile and link the following sample program to the LuaSocket object files and the static LuaJIT library:
```
gcc -Wall -I/home/siffiejoe/.self/programs/luajit-2.1/include/luajit-2.1 -o foo foo.c *.o /home/siffiejoe/.self/programs/luajit-2.1/lib/libluajit-5.1.a -ldl -lm
```
Result:
```
/home/siffiejoe/.self/programs/luajit-2.1/lib/libluajit-5.1.a(ljamalg.o): In function `luaL_testudata':
ljamalg.c:(.text+0x4a150): multiple definition of `luaL_testudata'
compat.o:/home/siffiejoe/src/repos/luasocket/src/compat.c:23: first defined here
/home/siffiejoe/.self/programs/luajit-2.1/lib/libluajit-5.1.a(ljamalg.o): In function `luaL_setfuncs':
ljamalg.c:(.text+0x53ca0): multiple definition of `luaL_setfuncs'
compat.o:/home/siffiejoe/src/repos/luasocket/src/compat.c:7: first defined here
collect2: error: ld returned 1 exit status
```

Sample program:
```C
#include <stddef.h>
#include <stdlib.h>
#include <stdio.h>
#include <lua.h>
#include <lualib.h>
#include <lauxlib.h>

int luaopen_socket_core( lua_State* L );

int main( void ) {
  int rv = 0;
  int status = EXIT_FAILURE;
  lua_State* L = luaL_newstate();
  if( !L ) {
    fprintf( stderr, "Could not create Lua state!\n" );
    goto bailout;
  }
  lua_pushcfunction( L, luaopen_socket_core );
  lua_pushliteral( L, "socket.core" );
  rv = lua_pcall( L, 1, 1, 0 );
  if( rv ) {
    fprintf( stderr, "socket.core: %s\n", lua_tostring( L, -1 ) );
    goto bailout;
  }
  status = EXIT_SUCCESS;
bailout:
  if( L )
    lua_close( L );
  return status;
}
```

This pull request linked above is one trivial way to fix this.

--- a/src/compat.h
+++ b/src/compat.h
@@ -5,6 +5,8 @@
 #include "lauxlib.h"
 
 #if LUA_VERSION_NUM==501
+#define luaL_setfuncs socket_setfuncs
+#define luaL_testudata socket_testudata
 void luaL_setfuncs (lua_State *L, const luaL_Reg *l, int nup);
 void *luaL_testudata ( lua_State *L, int arg, const char *tname);
 #endif

